packages/moonstream-components/src/core/providers/Web3Provider/index.tsx

```diff
@@ -300,13 +318,10 @@ const Web3Provider = ({ children }: { children: JSX.Element }) => {
 
   // When chainId or web3 or targetChain changes -> update button state
   React.useLayoutEffect(() => {
-    if (web3.currentProvider && chainId && targetChain?.chainId) {
+    console.log(chainId, targetChain?.chainId, account, isKnownChain(chainId));
+    if (web3.currentProvider && chainId && targetChain?.chainId && account) {
       if (isKnownChain(chainId)) {
-        if (chainId === targetChain?.chainId) {
-          setButtonText(WALLET_STATES.CONNECTED);
-        } else {
-          setButtonText(`Please select ${targetChain?.name}`);
```

Comment:
This feels like we are already paying a price for coupling a user message with our wallet state. Isn't the previous error message better than the non-specific one?

Reply:
@kellan-simiotics since we have chosen policy of not enforcing chain selection this becomes outdated - chain cannot be wrong, it can be either supported, or not supported. If user does not change his chain